src/store/Workspaces/devWorkspaces/index.ts

```diff
@@ -318,20 +321,35 @@ export const reducer: Reducer<State> = (state: State | undefined, action: KnownA
 function onStatusUpdateReceived(
   workspace: IDevWorkspace,
   dispatch: ThunkDispatch<State, undefined, KnownAction>,
-  message: any) {
-  let status: string;
-  if (message.error) {
+  statusUpdate: IStatusUpdate) {
+  let status: string | undefined;
+  if (statusUpdate.error) {
     const workspacesLogs = new Map<string, string[]>();
-    workspacesLogs.set(workspace.status.devworkspaceId, [`Error: Failed to run the workspace: "${message.error}"`]);
+    workspacesLogs.set(workspace.status.devworkspaceId, [`Error: Failed to run the workspace: "${statusUpdate.error}"`]);
     dispatch({
       type: 'DEV_UPDATE_WORKSPACES_LOGS',
       workspacesLogs,
     });
     status = WorkspaceStatus[WorkspaceStatus.ERROR];
   } else {
-    status = message.status;
+    if (statusUpdate.message) {
+      const workspacesLogs = new Map<string, string[]>();
+
+      /**
+       * Don't add in messages with no workspaces id or with stopped or stopping messages. The stopped and stopping messages
+       * only appear because we initially create a stopped devworkspace, add in devworkspace templates, and then start the devworkspace
+       */
+      if (workspace.status.devworkspaceId !== '' && workspace.status.message !== 'Stopped' && workspace.status.message !== 'Stopping') {
```

Comment:
I think we have some confusions with DevWorkspaceStatus 
```
export enum DevWorkspaceStatus {
  FAILED = 1,
  RUNNING = 2
}
```
https://github.com/eclipse/che-dashboard/blob/devworkspace-logs/src/services/helpers/types.ts#L39

It seems DevWorkspaceStatus is partly implemented and it is hard to use. For example:

```
...
        status = status.toUpperCase();
        if (status === DevWorkspaceStatus[DevWorkspaceStatus.FAILED]) {
          status = WorkspaceStatus[WorkspaceStatus.ERROR];
        }
...
```
https://github.com/eclipse/che-dashboard/blob/devworkspace-logs/src/services/helpers/workspaceAdapter.ts#L127



 It could be better to modify DevWorkspaceStatus:
```
export enum DevWorkspaceStatus {
  FAILED = 'Failed',
  RUNNING = 'Running',
  STOPPING = 'Stopping',
  STOPPED = 'Stopped',
}
```

In this case it will be:
```
        if (status === DevWorkspaceStatus.FAILED) {
          status = WorkspaceStatus[WorkspaceStatus.ERROR];
        }
```
instead of 
```
        status = status.toUpperCase();
        if (status === DevWorkspaceStatus[DevWorkspaceStatus.FAILED]) {
          status = WorkspaceStatus[WorkspaceStatus.ERROR];
        }
```

and 

```
...
workspace.status.message !== DevWorkspaceStatus.STOPPED && workspace.status.message !== DevWorkspaceStatus.STOPPING
...
```
instead of 
```
...
workspace.status.message !== 'Stopped' && workspace.status.message !== 'Stopping'
...
```

WDYT?